src/kakarot/accounts/eoa/library.cairo

```diff
@@ -140,8 +140,22 @@ namespace ExternallyOwnedAccount {
             assert [current_tx_calldata] = payload_len;
             assert offset = 1;
             assert selector = DEPLOY_CONTRACT_ACCOUNT;
+            tempvar syscall_ptr = syscall_ptr;
```

Comment:
Well, one issue is that `IKakarot.execute_at_address` returns:
```
  stack_accesses_len: felt,
  stack_accesses: felt*,
  stack_len: felt,
  memory_accesses_len: felt,
  memory_accesses: felt*,
  memory_bytes_len: felt,
  evm_contract_address: felt,
  starknet_contract_address: felt,
  return_data_len: felt,
  return_data: felt*,
  gas_used: felt,
```
and not just a `res` pointer. Which is kinda annoying....

Reply:
indeed, but can't it be `let res = IKakarot.execute_at_address` and `res` is a tuple, which is a`felt*` or can be casted as such, right?

Reply:
Tbh I don't think that works when using interfaces, but maybe @greged93 will prove me wrong :D

Reply:
will let you guys know :)

Reply:
tried your solution @ClementWalter, unfortunately it seems like you can't cast the return from `execute_at_address` or `deploy_contract_account`, experimented with the following:
- `cast(res, felt*)`: can't cast a tuple to a felt
- `cast(&res, felt*): ap may only be used in an expression of the form [ap + <const>]
- using `&res`: same as above

Tried locally on a simple pure Cairo program and it works, thus probably fails here because of the interface call as mentioned by @TotalPizza. Let me know if you have another solution, otherwise will keep the current structure.

Reply:
For now I left it how it was (while adding the check on `amount == 0`), couldn't figure out another way

Reply:
Found a fix @ClementWalter

Reply:
Found a fix @ClementWalter